### DISCUSSION TITLE
Disabled follow points subsetting

### DISCUSSION THREAD

**Discussion by o-dasher** [ppy/osu#20388] [Ideas]
This discussion aims to try to find the best way that we should implement the option to disable follow points with Flashlight mod which was bring as a "requirement" by #20373.

the way it was being handled by the mentioned pull request is bad because this option could be useful for a variety of mods rather than just Flashlight, this means that there would be a big amount of code repetition and other problems, just to handle one "setting", which could potentially be a pita in a near future.

That's why we must come in a consensus on which path should be taken for making this change possible.

Here are my two cents.

**Difficulty Adjust mod**
With this way we could make a setting in the DA mod which disables the follow points, this allows for it to be passed for the server straightforward, and not too much to mess with.

**Skin based**
When your skin has no follow points this would mark your play as made without them, this information could be attached to the score i suppose.
I kind of disagree with this since it requires users messing with their skin, which is another task, and brings even more complexity for scores flow. it also does not give a hint that this may potentially affect pp values later on. and besides that it also wouldn't give a hint by checking the mods on leaderboards that a given score was done with follow points disabled, unless we do more needless complexity including UI changes just for that.

**A side panel containing mod sub settings**
This one would essentially be a side panel that includes all settings that may affect multiple mods rather than one, since disabling follow points fits into this category, it would make sense for it to be there, it would require a lot of rethinking of how things works, and how it should be displayed on scores and is much more complex than some other ways though.

**A Disable Forecasting mod**
Pretty much my personal pick, this mod would essentially just serve as a quick way to disable follow points without any hassle being bring to the user. this mod could potentially be turned on by default when there aren't follow points on the user skin to begin with.

**Comment by ghost** [comment]
I say skin based. I don't like the idea of adjusting a score based on whether or not followpoints are enabled, since it's already skinned out in many skins. If you want followpoints to affect scores, would something similar be implemented for combo colors and combo numbers? How would this be detected for the skins that skin those out too?

**Reply by peppy** [reply]
Having as a mod would allow it to be saved to replays, and can be done without a difficulty modifier. Do you have any reasoning against this approach?

**Reply by ghost** [reply]
Not in particular, mainly just that I don't believe that follow points being invisible would warrant a mod. I don't think it affects gameplay in any meaningful way.

**Reply by Feodor0090** [reply]
It affects. You don't need to understand in which order notes are placed, just move the cursor along the lines. This is even more noticeable with skins which use lines instead of dots or with FL/HD/TC/FreezeFrame.

**Reply by peppy** [reply]
I kind of agree having a mod just for this is likely a bit noisy. Maybe the mod can also disable other elements if any are identified (with each being a toggle if required).

The main goal is grouping this away from something like flashlight mod.

**Reply by mk56-spn** [reply]
> I kind of agree having a mod just for this is likely a bit noisy. Maybe the mod can also disable other elements if any are identified (with each being a toggle if required).
> 
> The main goal is grouping this away from something like flashlight mod.

Agreed. How would you feel about moving "hide only approach " out of hidden subsettings. It always felt slightly off as a hidden subsetting to me. And if we're gonna go forward with some difficulty affecting  skin parts being togglable I feel it would be better grouped in this hypothetical new mod.  Although I don't know how feasible this is for you from a back end pov with having to shift scores parts around

**Reply by o-dasher** [reply]
I would suggest the mod naming to be `Visual Adjusts`. 
I am not sure about the settings to implement together with removing follow points, but yeah hide approach circles would be nice here rather than a HD option.

**Reply by o-dasher** [reply]
Maybe adding remove combo numbers and colorings as it was mentioned previously.

**Comment by ghost** [comment]
Quick question: are we planning on making stuff like this (e.g. disabled follow points) affect difficulty and pp calculation?

**Reply by o-dasher** [reply]
Yes, this is the intention of the referenced post.

**Reply by ghost** [reply]
I actually don't see any opinions regarding this from the community, whether for or against. Could you link the relevant messages?

**Reply by o-dasher** [reply]
Won't links cuz i am on mobile right now but it is a simple as, flashlight without follow points is objectively harder than flashlight with follow points regarding aiming outside of the visible area so it should reward more pp

**Reply by peppy** [reply]
Keep in mind this is up to the community and SR/PP teams to decide, not a single person.

**Reply by mk56-spn** [reply]
well, my post was merely a suggestion, but i will say, regardless of whether it gives pp, and if thats now, in a year , or 5, or 10. has rather little bearing on this being a good mod IMO. considering pp development has almost always been, to put it mildly, lethargic, it is safe to assume that the sheer amount of difficulty affecting mods in lazer and the exponential amount of combination of these that themselves radically change gp, that PP wont be accounting for even like 20% of what it potentially should/ could for the forseeable future, meaning i dont think it should be the axiom upon whether this option depends since its fun to see who managed it with and without the added difficulty regardless
